File: BackEnd/progetto IdS/CasottoIdS/src/main/java/it/unicam/cs/CasottoIdS/models/SlotData.java
package it.unicam.cs.CasottoIdS.models;

import java.util.Date;
import java.util.Objects;

public class SlotData {

    private Giorno durata;
    private Date data;

    public SlotData(Giorno durata, Date data) {
        this.durata = durata;
        this.data = data;
    }

    public Giorno getDurata() {
        return durata;
    }

    public void setDurata(Giorno durata) {
        this.durata = durata;
    }

    public Date getData() {
        return data;
    }

    public void setData(Date data) {
        this.data = data;
    }

    @Override
    public String toString() {
        return "SlotData{" +
                "durata=" + durata +
                ", data=" + data +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        SlotData slotData = (SlotData) o;
        return durata == slotData.durata && data.equals(slotData.data);
    }

    }

File: FrontEnd/Casotto/lib/models/SlotData.dart
import 'Giorno.dart';

class SlotData {
  late Giorno _durata;
  late DateTime _data;

  SlotData(Giorno durata, DateTime data) {
    this._durata = durata;
    this._data = data;
  }

  Giorno getDurata() {
    return this._durata;
  }

  void setDurata(Giorno durata) {
    this._durata = durata;
  }

  DateTime getData() {
    return this._data;
  }

  void setData(DateTime data) {
    this._data = data;
  }

  String getDurataString() {
    switch (this._durata) {
      case Giorno.MATTINA:
        return "MATTINA";
      case Giorno.POMERIGGIO:
        return "POMERIGGIO";
    }
  }

  String getDataString() {
    return "${this._data.day}/${this._data.month}/${this._data.year}";
  }

  @override
  String toString() {
    return "SlotData{" +
        " durata= " +
        _durata.toString() +
        ", data= " +
        _data.toString() +
        "}";
  }
}
